Fix tournament chat potentially crashing due to null current match

cdwcgt opened the pull request:
- related https://github.com/ppy/osu/pull/23890

When no bracket is selected, `ladderInfo.CurrentMatch.Value` is null.
and `LadderInfo.cs` have `#nullable disable` so there is no build warning

cdwcgt committed:
remove `#nullable disable` in `LadderInfo`

cdwcgt committed:
null guard for Current Match in MatchMessage

bdach commented:
Nullability was disabled again on `LadderInfo` because it causes issues all over tournament client. See code quality inspections.

If you're going to be fixing, then fix everything.

cdwcgt commented:
Ok, I'll leave `#nullable disable` for now

cdwcgt committed:
Revert "remove `#nullable disable` in `LadderInfo`"

This reverts commit 0dced4610035b07c48856324314c3be5a18af92f.

bdach committed:
Use alternative guard

Early-returning from ctors feels pretty bad. Also saves on some
nested accesses.

bdach committed:
Remove no-longer-needed BDL `permitNulls` spec

bdach committed:
Rename test step

bdach committed:
Merge branch 'master' into chatdisplay-fix
